master/buildbot/db/compression/zstd.py

```diff
@@ -29,10 +30,36 @@ class ZStdCompressor(CompressorInterface):
     name = "zstd"
     available = HAS_ZSTD
 
+    COMPRESS_LEVEL = 9
+
     @staticmethod
     def dumps(data: bytes) -> bytes:
-        return zstandard.compress(data, level=9)
+        return zstandard.compress(data, level=ZStdCompressor.COMPRESS_LEVEL)
 
     @staticmethod
     def read(data: bytes) -> bytes:
-        return zstandard.decompress(data)
+        # not too sure why, but zstandard.decompress do not work with
+        # data compressed with `ZstdCompressionObj`
```

Comment:
I think this is pretty serious problem and indicates some kind of a bug. Could we compare the bytes emitted by ZstdCompressionObj and compare them to what zstandard.compress emits?

Reply:
Sure, I'll dig into this more to at least leave a better comment.
I only did a quick search on this in zstandard issue which did not lead to a concrete reason for this.

Reply:
Found an issue that explained the problem, updated the comment with a link.